lib/block_view_derived_key_test.go

```diff
@@ -2844,6 +3080,86 @@ func TestAuthorizedDerivedKeyWithTransactionLimitsHardcore(t *testing.T) {
 		require.Contains(err.Error(), RuleErrorDerivedKeyCreatorCoinOperationNotAuthorized)
 	}
 
+	// Randomly try changing the spending limit on the derived key to an unlimited key.
+	{
+		// Get the mempool's utxoview and get the derived key bytes.
+		utxoView, err := mempool.GetAugmentedUniversalView()
+		require.NoError(err)
+		derivedPub := derived0Priv.PubKey()
+		derivedPubBytes := derivedPub.SerializeCompressed()
+		require.NoError(err)
+
+		// Persist the existing spending limit on the derived key.
+		prevDerivedKeyEntry := utxoView._getDerivedKeyMappingForOwner(m0PkBytes, derivedPubBytes)
+		require.NotNil(prevDerivedKeyEntry)
+		require.Equal(false, prevDerivedKeyEntry.isDeleted)
+		prevTransactionSpendingLimit := &TransactionSpendingLimit{}
+		prevTransactionSpendingLimitBytes, err := prevDerivedKeyEntry.TransactionSpendingLimitTracker.ToBytes(1)
+		rr := bytes.NewReader(prevTransactionSpendingLimitBytes)
+		err = prevTransactionSpendingLimit.FromBytes(1, rr)
+		require.NoError(err)
+
+		// Unlimited spending limit.
+		transactionSpendingLimit = &TransactionSpendingLimit{
+			GlobalDESOLimit:              0,
+			TransactionCountLimitMap:     make(map[TxnType]uint64),
+			CreatorCoinOperationLimitMap: make(map[CreatorCoinOperationLimitKey]uint64),
+			DAOCoinOperationLimitMap:     make(map[DAOCoinOperationLimitKey]uint64),
+			NFTOperationLimitMap:         make(map[NFTOperationLimitKey]uint64),
+			IsUnlimited:                  true,
+		}
```

Comment:
define a var for this unlimited spending limit struct?

Reply:
We only use this like twice in the entire file, I personally prefer it to be more descriptive.